### PR TITLE
Add semgrep-pro in semgrep docker image by default

### PR DESCRIPTION
**Pull request by aryx**
Closes EA-5

test plan:
run the
https://github.com/semgrep/semgrep/actions/workflows/build-test-docker.yaml
workflow manually, with canary-pro for name of the output
and check if this canary-pro has indeed semgrep-core-proprietary in it

### CODE REVIEW

**Comment by aryx** [comment]
job triggered manually here: https://github.com/semgrep/semgrep/actions/runs/7698368490

File: Makefile
Comment (IagoAbal) [inline Makefile:131]: So this would be a way to generate an OSS-only Docker image, but the one we will be publishing to Docker Hub will contain the Pro binary?

Comment (aryx) [reply Makefile:131]: correct

**Comment by aryx** [comment]
try 2: https://github.com/semgrep/semgrep/actions/runs/7699362232

**Comment by IagoAbal** [comment]
Need a changelog?

File: Dockerfile
Comment (IagoAbal) [inline Dockerfile]: @tpetr suggested something like `[[ -z "${SEMGREP_APP_TOKEN}" ]] || semgrep install-semgrep-pro` so that the image would build even if `{SEMGREP_APP_TOKEN` isn't available. IMO this is a minor issue, if need be I'd better ship this one fast. But if we can make `docker build` just work easily then that's even better of course.

**Review by tpetr — CHANGES_REQUESTED** [review]
2 things I'm checking on before I feel comfortable merging:
1. make sure the act of us installing pro engine isn't accidentally laying down sensitive files in the docker image ✅ 
2. make sure pro engine is actually installed on linux/arm64 (see depot build failures)

**Comment by aryx** [comment]
still the error in the build-docker-arm64:
``` > [semgrep-cli 1/2] RUN --mount=type=secret,id=SEMGREP_APP_TOKEN SEMGREP_APP_TOKEN=$(cat /run/secrets/SEMGREP_APP_TOKEN) semgrep install-semgrep-pro --debug:
#34 6.239            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
#34 6.239   File "/usr/local/lib/python3.11/subprocess.py", line 548, in run
#34 6.239     with Popen(*popenargs, **kwargs) as process:
#34 6.239          ^^^^^^^^^^^^^^^^^^^^^^^^^^^
#34 6.239   File "/usr/local/lib/python3.11/subprocess.py", line 1026, in __init__
#34 6.239     self._execute_child(args, executable, preexec_fn, close_fds,
#34 6.239   File "/usr/local/lib/python3.11/subprocess.py", line 1950, in _execute_child
#34 6.239     raise child_exception_type(errno_num, err_msg, err_filename)
#34 6.239 OSError: [Errno 8] Exec format error: '/usr/local/bin/semgrep-core-proprietary.tmp_download'
#34 6.239 Not sending pseudonymous metrics since metrics are configured to OFF and registry usage is False
------
```

**Comment by aryx** [comment]
I think I need to update the code of pysemgrep install-semgrep-pro probably

**Comment by aryx** [comment]
@brandonspark wrote this code I think; we probably need to better look for the arch and detect the linux-arm64 (and not just linux)

**Comment by aryx** [comment]
I'll do that monday if nobody does it before me.

File: cli/src/semgrep/commands/install.py
Comment (tpetr) [inline cli/src/semgrep/commands/install.py]: `platform.machine()` returns the value of `uname -m`. I don't think checking for "arm" is sufficient here because there are plenty of ARM machines that return `aarch64` (see https://en.wikipedia.org/wiki/Uname for reference). Checking for `arm` or `aarch` feels like a reasonable compromise.

**Comment by aryx** [comment]
progress but still error:
```
p-cli 1/2] RUN --mount=type=secret,id=SEMGREP_APP_TOKEN SEMGREP_APP_TOKEN=$(cat /run/secrets/SEMGREP_APP_TOKEN) semgrep install-semgrep-pro --debug:
#34 0.831 Semgrep Pro Engine will be installed in /usr/local/bin/semgrep-core-proprietary
#34 0.831 platform is linux
#34 0.831 downloading from https://semgrep.dev/api/agent/deployments/deepbinary/linux-arm64?version=1.59.1
#34 2.624 Logged in deployment does not have access to Semgrep Pro Engine
#34 2.624 Visit https://semgrep.dev/products/pro-engine/ for more information.
#34 2.625 Not sending pseudonymous metrics since metrics are configured to OFF and registry usage is False
------

Build Summary: https://depot.dev/orgs/9ks3jwp44z/projects/fhmxj6w9z8/builds/lbrw6bl9ts
Error: failed to solve: process "/bin/sh -c SEMGREP_APP_TOKEN=$(cat /run/secrets/SEMGREP_APP_TOKEN) semgrep install-semgrep-pro --debug" did not complete successfully: exit code: 2
Warning: falling back to buildx: Error: failed with: Error: failed to solve: process "/bin/sh -c SEMGREP_APP_TOKEN=$(cat /run/secrets/SEMGREP_APP_TOKEN) semgrep install-semgrep-pro --debug" did not complete successfully: exit code: 2

```

**Comment by aryx** [comment]
#34 2.624 Logged in deployment does not have access to Semgrep Pro Engine

lol

**Comment by aryx** [comment]
seems to work now:
```
#34 [semgrep-cli 1/2] RUN --mount=type=secret,id=SEMGREP_APP_TOKEN SEMGREP_APP_TOKEN=$(cat /run/secrets/SEMGREP_APP_TOKEN) semgrep install-semgrep-pro --debug
#34 1.656 Semgrep Pro Engine will be installed in /usr/local/bin/semgrep-core-proprietary
#34 1.656 platform is linux
#34 1.656 downloading from https://semgrep.dev/api/agent/deployments/deepbinary/linux-arm64?version=1.59.1
#34 8.256 Downloading... ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 180.2/180.2 MB 32.8 MB/s 0:00:00
#34 8.327 Successfully installed Semgrep Pro Engine (version b609443)!
#34 8.327 Not sending pseudonymous metrics since metrics are configured to OFF and registry usage is False
#34 DONE 8.5s
```
@tpetr can I merge this? I think we've addressed your 2 top requests.

**Comment by tpetr** [comment]
@aryx taking one last look

**Review by tpetr — APPROVED** [review]
it feels a little bad that I can no longer run a vanilla `docker build` on this repo anymore, but my original suggestion to first check the env var didn't work when I tested locally and I don't want to hold up this PR anymore, so let's try improving that in a future PR

**Pull request merged**
